staging/src/k8s.io/apiserver/pkg/endpoints/filters/tenantInfo.go

```diff
@@ -60,16 +60,16 @@ func WithTenantInfo(handler http.Handler) http.Handler {
 
 // normalizeObjectTenant returns the tenant of the object based on what the user tenant is.
 func normalizeTenant(userTenant, objectTenant string) (string, error) {
```

Comment:
We actually only change objectTenant in one situation, and keep objectTenant untouched in all other situations.  Suggest refactoring the codes of normalizeTenant() to make our intention explicit:

//the only case we change objectTenant: tenant info is not in URL and the request is issued by a non-system tenant
If objectTenant==metav1.TenantNone && userTenant != TenantSystem {
  return userTenant
}

return objectTenant